File: app/src/main/java/com/ruzhan/recyclerviewitemanimation/OneSlideActivity.java
package com.ruzhan.recyclerviewitemanimation;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.ruzhan.recyclerviewitemanimation.adapter.OneSlideAdapter;

import java.util.ArrayList;
import java.util.List;

public class OneSlideActivity extends AppCompatActivity implements View.OnClickListener {

    private RecyclerView mRecyclerView;
    private OneSlideAdapter mOneSlideAdapter;
    private TextView mRightTV;

    private List<String> mData;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.one_slide_activity);

        addTestData();

        initView();
        initData();
        initListener();
    }

    private void addTestData() {
        mData = new ArrayList<>();
        for (int x = 0; x < 66; x++) {
            mData.add("" + x);
        }
    }

    private void initData() {
        mOneSlideAdapter = new OneSlideAdapter();
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        mRecyclerView.setAdapter(mOneSlideAdapter);
        mOneSlideAdapter.setData(mData);
    }

    private void initListener() {
        mRightTV.setOnClickListener(this);
    }

    private void initView() {
        mRecyclerView = findViewById(R.id.rcv_root);
        mRightTV = findViewById(R.id.right_tv);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.right_tv:
                editItems();
                break;
        }
    }

    private void editItems() {
        if ("CLOSE".equals(mRightTV.getText().toString())) {
            mRightTV.setText("OPEN");
            mOneSlideAdapter.slideOpen();
        } else if ("OPEN".equals(mRightTV.getText().toString())) {
            mRightTV.setText("CLOSE");
            mOneSlideAdapter.slideClose();
        }
    }
}

File: app/src/main/java/com/ruzhan/recyclerviewitemanimation/OneSlideActivity.kt
package com.ruzhan.recyclerviewitemanimation

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.ruzhan.recyclerviewitemanimation.adapter.OneSlideAdapter
import kotlinx.android.synthetic.main.one_slide_activity.*

class OneSlideActivity : AppCompatActivity() {

    companion object {
        private const val OPEN = "OPEN"
        private const val CLOSE = "CLOSE"
    }

    private var oneSlideAdapter = OneSlideAdapter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.one_slide_activity)
        initData()
        initListener()
    }

    private fun initData() {
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = oneSlideAdapter
    }

    private fun initListener() {
        rightTv.setOnClickListener {
            editItems()
        }
    }

    private fun editItems() {
        if (CLOSE == rightTv.text.toString()) {
            rightTv.text = OPEN
            oneSlideAdapter.slideOpen()
        } else if (OPEN == rightTv.text.toString()) {
            rightTv.text = CLOSE
            oneSlideAdapter.slideClose()
        }
    }
}
